### PR TITLE
SPPEEEED dial

### PR DESCRIPTION
**Pull request by alex-power**

### CODE REVIEW

**Comment by alex-power** [comment]
image

**Comment by clayplumridge** [comment]
Looks like you've got some linter errors. If you install all of the recommended vscode extensions you should be autofixing them on file save. If you don't want to for some reason, run `npm run lint:fix`

File: src/render/theme.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const theme = createMuiTheme({
     },
     props: {
         MuiButton: {
-            disableRipple: true
+            disableRipple: false
```
Comment (clayplumridge) [inline src/render/theme.ts]: You can just remove the entire `props` block. This merges with the base dark theme from MUI. Another cool application of the spread operator: `{ ...themeDefaults, ...themeOverrides }`

File: src/render/views/CreateEncounterView.tsx [hunk 2]
```diff
@@ -22,6 +29,15 @@ const useStyles = makeStyles(theme =>
         },
         notLastButton: {
             marginRight: theme.spacing(1)
+        },
```
Comment (clayplumridge) [inline src/render/views/CreateEncounterView.tsx]: Looks like most of these styles aren't being used anymore; can probably remove them

File: src/render/views/CreateEncounterView.tsx [hunk 3]
```diff
@@ -30,26 +46,54 @@ export const CreateEncounterView: React.FC<{}> = () => {
     const viewManager = getViewManager();
     const encounterManager = getEncounterManager();
     const [name, setName] = useObservable("");
+    const [actors, setActors] = useObservableArray<Actor>([]);
+    const [speedDialOpen, setSpeedDialOpen] = useObservable(false);
+    const styles = useStyles();
+    const speedDialActions = [
```
Comment (clayplumridge) [inline src/render/views/CreateEncounterView.tsx]: You could also define this outside of the component definition as top-level static state so we don't re-initialize a bunch of objects on every render of this component. I do this with most maps and arrays like this one. The performance cost is low but can add up on complex pages.

File: src/render/views/CreateEncounterView.tsx [hunk 4]
```diff
@@ -30,26 +46,54 @@ export const CreateEncounterView: React.FC<{}> = () => {
...
+        // ]);
+        //actors.concat(db.getActors(playercharacters));
+    };
+
     const cancel = () => {
         viewManager.onViewChanged(View.EncounterManagement);
     };
 
-    const styles = useStyles();
+    const handleSpeedDialClose = () => {
```
Comment (clayplumridge) [inline src/render/views/CreateEncounterView.tsx]: These are short enough that you can just inline them in the `SpeedDial` itself as `onOpen={() => setSpeedDialOpen(true)}` etc.

File: src/render/views/CreateEncounterView.tsx [hunk 5]
```diff
@@ -78,6 +122,32 @@ export const CreateEncounterView: React.FC<{}> = () => {
                     <Typography>Cancel</Typography>
                 </Button>
             </Box>
+            <Observer observed={{ speedDialOpen }}>
+                {({ speedDialOpen }) => {
+                    return (
+                        <SpeedDial
+                            className={styles.fab}
+                            color="primary"
+                            ariaLabel="Add Character"
```
Comment (clayplumridge) [inline src/render/views/CreateEncounterView.tsx]: Let's leave all `aria` props out for now. If we want to do an accessibility pass later we definitely can (and likely should) but I'm not sure what the right usage pattern is here so I'd rather leave it alone.

File: src/render/views/CreateEncounterView.tsx [hunk 6]
```diff
@@ -78,6 +122,32 @@ export const CreateEncounterView: React.FC<{}> = () => {
...
+                            onClick={() => console.log("hello")}
```
Comment (clayplumridge) [inline src/render/views/CreateEncounterView.tsx]: Left in by mistake?

**Review by clayplumridge — CHANGES_REQUESTED** [review]

File: src/render/views/CreateEncounterView.tsx (hunk 3 above)
Comment (clayplumridge) [reply src/render/views/CreateEncounterView.tsx]: The other option here is to use `const speedDialActions = React.useMemo(() => [ ... ]);` which will memoize the call so it doesn't recalculate it every render. Not super partial here tbh.

Comment (clayplumridge) [reply src/render/views/CreateEncounterView.tsx]: I actually think I'd prefer `React.useMemo` after thinking on it a bit here. It encapsulates the info we care about inside the component which is clearer and delays the logic until first render.

**Review by clayplumridge — APPROVED** [review]

**Pull request merged**